test/e2e/dns.go

```diff
@@ -0,0 +1,409 @@
+							},
+							VolumeMounts: []corev1.VolumeMount{
+								{
+									Name:      "resolv-conf",
+									MountPath: "/tmp/resolv.conf",
+									ReadOnly:  true,
+								},
+							},
+						},
+					},
+					Volumes: []corev1.Volume{
+						{
+							Name: "resolv-conf",
+							VolumeSource: corev1.VolumeSource{
+								HostPath: &corev1.HostPathVolumeSource{
+									Path: "/etc/resolv.conf",
+									Type: &hpt,
+								},
+							},
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyOnFailure,
+				},
+			},
+		},
+	}
+	_, err := cli.BatchV1().Jobs(namespace).Create(ctx, job, metav1.CreateOptions{})
+	return err
+}
+
+func deleteResolvConfJob(ctx context.Context, cli kubernetes.Interface, nodeName string, namespace string) error {
+	dpb := metav1.DeletePropagationBackground
+	return cli.BatchV1().Jobs(namespace).Delete(ctx, resolvConfJobName(nodeName), metav1.DeleteOptions{
+		PropagationPolicy: &dpb,
+	})
+}
+
+func verifyResolvConf(ctx context.Context, cli kubernetes.Interface, nodeName string, namespace string, nodeIp string) error {
+	jobName := resolvConfJobName(nodeName)
+	podList, err := cli.CoreV1().Pods(namespace).List(ctx, metav1.ListOptions{
+		LabelSelector: fmt.Sprintf("job-name=%s", jobName),
+	})
+	if err != nil {
+		return err
+	}
+	if len(podList.Items) != 1 {
+		return fmt.Errorf("found %v Pods associated with the Job, but there should be exactly 1", len(podList.Items))
+	}
+
+	podName := podList.Items[0].ObjectMeta.Name
+	tailLines := int64(10)
+	podLogOptions := &corev1.PodLogOptions{
+		Container: resolvConfContainerName,
+		Follow:    false,
+		TailLines: &tailLines,
+	}
+	podLogRequest := cli.CoreV1().Pods(namespace).GetLogs(podName, podLogOptions)
+	stream, err := podLogRequest.Stream(ctx)
+	if err != nil {
+		return err
+	}
+	defer stream.Close()
+
+	logs := []string{}
+	for {
+		buf := make([]byte, 2000)
+		numBytes, err := stream.Read(buf)
+		if numBytes == 0 {
+			break
+		}
+		if err == io.EOF {
+			break
+		}
+		if err != nil {
+			return err
+		}
+		logs = append(logs, string(buf[:numBytes]))
+	}
+
+	log := strings.Join(logs, "\n")
+	if log == "" {
+		return fmt.Errorf("pod didn't have any logs")
+	}
+
+	nameserverLine := nameserverRegex.FindString(log)
+	if nameserverLine == "" {
+		return fmt.Errorf("didn't find nameserver in resolv.conf")
+	}
+
+	_, nameserverIp, _ := strings.Cut(nameserverLine, " ")
+	if nameserverIp != nodeIp {
+		return fmt.Errorf("nameserver specified in resolv.conf does not match node's IP address")
+	}
+
+	return nil
+}
+
+func toggleAcceleratedNetworking(ctx context.Context, interfaces network.InterfacesClient, clusterResourceGroup string, nodeName string, enabled bool) error {
+	nic, err := interfaces.Get(ctx, clusterResourceGroup, nicName(nodeName), "")
+	if err != nil {
+		return err
+	}
+
+	nic.EnableAcceleratedNetworking = to.BoolPtr(enabled)
+	err = clients.Interfaces.CreateOrUpdateAndWait(ctx, clusterResourceGroup, nicName(nodeName), nic)
+	return err
```

Comment:
Instead of doing the `sleep` after each toggle in the tests, can we just sleep here in the function itself, and/or leverage `Eventually` or something else in ginkgo to handle the wait?

Reply:
If we are sleeping for the purposes of waiting for a nic update, and we are updating all of the nics, I think it makes sense to sleep for each one. Do we have an idea of how much time it takes to let the change settle on a per-nic basis, or are we just making a ballpark estimate to avoid flakes?

Reply:
The time I chose is a ballpark estimate to avoid flakes, unfortunately. I wanted to make the test resilient to flakes, I'm not sure how we'd determine exactly how long it takes for the change to truly take effect. I'm thinking that even if we `GET` the NIC and the field is updated, there might still be some stuff on the Network RP backend that hasn't happened yet.

What do we gain by sleeping for each one though? Assuming that it takes 5 seconds for a NIC update to settle, I can just iteratively update all three and then sleep for 5 seconds. If I sleep for 5 seconds after each update, I'm wasting 10 seconds in the middle.

Reply:
I think it's fine since we can only ballpark. It's not super ideal to have a sleep in a test but we might not have enough visibility on how long this will take in each region we plan on running this test